doc/src/tutorial/assumptions.rst

```diff
@@ -0,0 +1,1831 @@
+    >>> xpos.is_positive is S.true
+    False
+    >>> (xpos > 0) is True
+    False
+    >>> (xpos > 0) is S.true
+    True
+
+There is no general symbolic analogue of ``None`` in SymPy. In the cases where
+a low-level assumptions query gives ``None`` the symbolic query will result in
+an unevaluated symbolic ``Boolean`` (e.g, ``x > 0``).  We can use a symbolic
+``Boolean`` as part of a symbolic expression:
+
+    >>> from sympy import Piecewise
+    >>> p = Piecewise((1, x > 0), (2, True))
+    >>> p
+    Piecewise((1, x > 0), (2, True))
+    >>> p.subs(x, 3)
+    1
+
+Here ``p`` represents an expression that will be equal to ``1`` if ``x > 0``
+or otherwise it will be equal to ``2``. The unevaluated ``Boolean`` inequality
+``x > 0`` represents the condition for deciding the value of the expression
+symbolically. When we substitute a value for ``x`` the inequality will resolve
+to ``S.true`` and then the ``Piecewise`` can evaluate to ``1`` or ``2``.
+
+The same will not work when using a fuzzy-bool instead of a symbolic
+``Boolean``:
+
+    >>> p2 = Piecewise((1, x.is_positive), (2, True))
+    Traceback (most recent call last):
+    ...
+    TypeError: Second argument must be a Boolean, not `NoneType`
+
+The ``Piecewise`` can not use ``None`` as the condition because unlike the
+inequality ``x > 0`` it gives no information. With the inequality it is
+possible to decide in future if the condition might ``True`` or ``False``
+once a value for ``x`` is known. A value of ``None`` can not be used in that
+way so it is rejected.
+
+.. note:: We can use ``True`` in the ``Piecewise`` because ``True`` sympifies
+          to ``S.true``. Sympifying ``None`` just gives ``None`` again which
+          is not a valid symbolic SymPy object.
+
+There are many other symbolic ``Boolean`` types in SymPy. The same
+considerations about the differences between fuzzy bool and symbolic
+``Boolean`` apply to all other sympy ``Boolean`` types. Just to give a
+different example there is ``Contains`` which represents the statement that an
+object is contained in a set:
+
+    >>> from sympy import Reals, Contains
+    >>> x = Symbol('x', real=True)
+    >>> y = Symbol('y')
+    >>> Contains(x, Reals)
+    True
+    >>> Contains(y, Reals)
+    Contains(y, Reals)
+    >>> Contains(y, Reals).subs(y, 1)
+    True
+
+The Python operator corresponding to ``Contains`` is ``in``. A quirk of ``in``
+is that it can only evaluate to a ``bool`` (``True`` or ``False``) so if the
+result is indeterminate then an exception will be raised:
+
+    >>> from sympy import I
+    >>> 2 in Reals
+    True
+    >>> I in Reals
+    False
+    >>> x in Reals
+    True
+    >>> y in Reals
+    Traceback (most recent call last):
+    ...
+    TypeError: did not evaluate to a bool: (-oo < y) & (y < oo)
+
+The exception can be avoided by using ``Contains(x, Reals)`` or
+``Reals.contains(x)`` rather than ``x in Reals``.
+
+
+Three-valued logic with fuzzy bools
+===================================
```

Comment:
Perhaps this section and the one below it could go in its own document about fuzzy booleans

Reply:
Yes, maybe it could. The problem is where should any of these documents go though?

The document I've written here feels like too much for the tutorial but all we have are the tutorial and the module docs. The module docs don't feel like a great place for a lengthy discussion either:

https://docs.sympy.org/latest/modules/assumptions/index.html
https://docs.sympy.org/latest/modules/core.html#module-sympy.core.assumptions

Reply:
Another thing that would be good to have somewhere would be a glossary page for terms like "vanilla symbol" and "fuzzy bool". It would also be good to have a documented that lists all `is_*` properties and explains what they mean. There isn't a clear place for those either.

Reply:
We can create new rst files for long form guides like this. They can be linked to from the API docs. 

In general, I think we should try to organize our documentation using this system https://documentation.divio.com/.

Reply:
Yes that looks like a good structure. At the moment we only have two of the four quadrants.